rust_src/src/process.rs

```diff
@@ -318,4 +323,20 @@ pub fn set_process_query_on_exit_flag(process: LispObject, flag: LispObject) ->
     flag
 }
 
+/// Return the exit status of PROCESS or the signal number that killed it.
+/// If PROCESS has not yet exited or died, return 0.
+#[lisp_fn]
+pub fn process_exit_status(process: LispObject) -> LispObject {
+    let p_ref = process.as_process_or_error();
+    if p_ref.raw_status_new() {
+        unsafe { update_status(p_ref.as_ptr()) };
+    }
+    let status = LispObject::from_raw(p_ref.status);
+    if status.is_cons() {
+        status.xcons().cdr().xcons().car()
+    } else {
```

Comment:
Would this read any better if you used one of the iterators? Something like `iter_cars().nth(2).unwrap()`?

Reply:
This doesn't work.